data_structures/src/chain.rs

```diff
@@ -2099,6 +2101,137 @@ impl UnspentOutputsPool {
     }
 }
 
+/// List of unspent outputs that can be spent by this node
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct OwnUnspentOutputsPool {
+    /// Map of output pointer to timestamp
+    /// Those UTXOs have a timestamp value to avoid double spending
+    map: HashMap<OutputPointer, u64>,
+    /// Counter of UTXOs bigger than the minimum collateral
+    bigger_than_min_counter: usize,
+}
+
+impl OwnUnspentOutputsPool {
+    pub fn get<Q: ?Sized>(&self, k: &Q) -> Option<&u64>
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.get(k)
+    }
+
+    pub fn get_mut<Q: ?Sized>(&mut self, k: &Q) -> Option<&mut u64>
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.get_mut(k)
+    }
+
+    pub fn contains_key<Q: ?Sized>(&self, k: &Q) -> bool
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.contains_key(k)
+    }
+
+    pub fn insert(&mut self, k: OutputPointer, v: u64, bigger_than_min: bool) -> Option<u64> {
+        let old_ts = self.map.insert(k, v);
+        if bigger_than_min {
+            self.bigger_than_min_counter += 1;
+        }
+
+        old_ts
+    }
+
+    pub fn remove(&mut self, k: &OutputPointer, bigger_than_min: bool) -> Option<u64> {
+        let ts = self.map.remove(k);
+
+        if ts.is_some() && bigger_than_min {
+            self.bigger_than_min_counter -= 1;
+        }
+
+        ts
+    }
+
+    pub fn drain(&mut self) -> std::collections::hash_map::Drain<OutputPointer, u64> {
+        self.bigger_than_min_counter = 0;
+        self.map.drain()
+    }
+
+    pub fn iter(&self) -> std::collections::hash_map::Iter<OutputPointer, u64> {
+        self.map.iter()
+    }
+
+    pub fn keys(&self) -> std::collections::hash_map::Keys<OutputPointer, u64> {
+        self.map.keys()
+    }
+
+    pub fn bigger_than_min_counter(&self) -> usize {
+        self.bigger_than_min_counter
+    }
+
+    pub fn len(&self) -> usize {
+        self.map.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.map.is_empty()
+    }
+}
+
+/// Information about our own UTXOs
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UtxoInfo {
+    /// Vector of OutputPointer with their values, time_locks and if it is ready for collateral
+    pub utxos: Vec<(OutputPointer, u64, u64, bool)>,
```

Comment:
Can you create a struct for this field? This would improve the JSON-RPC API by giving the fileds a name:

```rust
struct UtxoMetadata {
    output_pointer: OutputPointer,
    value: u64,
    timelock: u64,
    ready_for_collateral: bool,   
}
```

```suggestion
    pub utxos: Vec<UtxoMetadata>,
```